packages/webdriverio/src/utils/interception/devtools.js

```diff
@@ -0,0 +1,182 @@
+                headers[name] = value
+                return headers
+            }, {})
+            const { requestId, request } = event
+
+            for (const mock of mocks) {
+                /**
+                 * match mock url
+                 */
+                if (!minimatch(request.url, mock.url)) {
+                    continue
+                }
+
+                /**
+                 * match filter options
+                 */
+                if (
+                    /**
+                     * HTTP method
+                     */
+                    (mock.filterOptions.method && mock.filterOptions.method.toLowerCase() !== request.method.toLowerCase()) ||
+                    (mock.filterOptions.headers && containsObject(responseHeaders, mock.filterOptions.headers))
+                ) {
+                    continue
+                }
+
+                const { body, base64Encoded } = await client.send(
+                    'Fetch.getResponseBody',
+                    { requestId }
+                )
+
+                request.body = base64Encoded ? atob(body) : body
+                request.body = responseHeaders['Content-Type'] && responseHeaders['Content-Type'].includes('application/json')
+                    ? JSON.parse(request.body)
+                    : request.body
+                mock.matches.push(request)
+
+                /**
+                 * no stubbing if no overwrites were defined
+                 */
+                if (mock.respondOverwrites.length === 0) {
+                    continue
+                }
+
+                const { errorReason, overwrite, params } = mock.respondOverwrites[0].sticky
+                    ? mock.respondOverwrites[0]
+                    : mock.respondOverwrites.shift()
+
+                /**
+                 * when response is modified
+                 */
+                if (overwrite) {
+                    let body = overwrite
+                    if (typeof overwrite === 'function') {
+                        body = await overwrite(request, client)
+                    }
+
+                    if (typeof body !== 'string') {
+                        body = JSON.stringify(body)
+                    }
+
+                    let responseCode = params.statusCode || event.responseStatusCode
+                    const responseHeaders = [
+                        ...event.responseHeaders,
+                        ...Object.entries(params.headers || {}).map(([key, value]) => { key, value })
+                    ]
+
+                    /**
+                     * check if local file and load it
+                     */
+                    const responseFilePath = path.isAbsolute(body) ? body : path.join(process.cwd(), body)
+                    if (fs.existsSync(responseFilePath) && canAccess(responseFilePath)) {
+                        body = fs.readFileSync(responseFilePath).toString()
+                    } else if (body.startsWith('http')) {
```

Comment:
can you please help understanding use case of this one?

Reply:
This allows to just replace a web resource with another web resource. This is helpful if your mock (e.g. a JS or CSS file) is already hosted somewhere on the internet, e.g.

```js
mock.respond('https://test.com/some-image.png')
```